### DISCUSSION TITLE
type change bytearray(4) to FLOAT32 and back.

### DISCUSSION THREAD

**Discussion by a-tarasov7** [micropython/micropython#12629] [Using MicroPython] [answered]
I can't find a way to easily convert
bytearray(4)
to FLOAT32
and back.
More precisely, not to convert, but to change the interpretation type of 4 bytes - in one case it is bytearray(4) and at the same time in another case it is FLOAT32.

I need to stream values in FLOAT32 via a serial SPI channel and the recipient must restore the sequence of bytes again in FLOAT32.

In practice, this is what is available in C through the “union” type, where members with different types share the same memory area.

There is something similar in the UCTYPES module.
But trying to get the FLOAT32 address results in an error.
The description states that for FLOAT32
uctypes.addressof(obj)
does not work.

**Comment by dlech** [comment]
Use `struct.pack()` and `struct.unpack()`.

**Comment by andrewleech** [comment, marked as answer]
`uctypes` appears to work perfectly for this.
``` python
>>> import uctypes
>>> data = bytearray(4)
>>> data
bytearray(b'\x00\x00\x00\x00')
>>> s = uctypes.struct(uctypes.addressof(data), {'f32': uctypes.FLOAT32})
>>> s.f32
0.0
>>> s.f32 = 1.4
>>> data
bytearray(b'33\xb3?')
```

**Reply by a-tarasov7** [reply]
yes, this is what was needed!
Thank you very much!
Best regards!
